src/Knp/Menu/Twig/Helper.php

```diff
@@ -142,4 +165,30 @@ private function castMenu($menu)
 
         return $menu;
     }
+
+    /**
+     * @param ItemInterface $item
+     *
+     * @return ItemInterface|null
+     */
+    private function retrieveCurrentItem(ItemInterface $item)
+    {
+        $currentItem = null;
+
+        if ($this->matcher->isCurrent($item)) {
```

Comment:
I think that in your case, you should configure only the deeper element to match this route actually